### ISSUE TITLE
Sender "chaining" does not work as advertised in the messenger component

### ISSUE THREAD

**Issue by BillyMorgan** [symfony/symfony#27908]
**Symfony version(s) affected**: 4.1.1

**Description**  
The messenger component documentation says that "A class of messages can also be routed to multiple senders by specifying a list", but this doesn't work. You get an exception stating that the SenderLocator "is a smaller service locator that only knows about the .... jobs".

**How to reproduce**  
The documented use case ought to reproduce it:
```
framework:
    messenger:
        routing:
            'My\Message\ToBeSentToTwoSenders': [amqp, audit]

```
**Possible Solution**  
Problem seems to be that the chained sender service that gets generated during the compiler pass is added to the global container, when it needs to be added to the service locator. Maybe?

**Comment by sroze** [comment]
Can you give us the exact exception (with its track trace) and the full
messenger configuration that you have?

Thank you :)

**Comment by BillyMorgan** [comment]
Messenger config:
```
framework:
    messenger:
        transports:
            # Uncomment the following line to enable a transport named "amqp"
            amqp: '%env(MESSENGER_TRANSPORT_DSN)%'
            amqp2: '%env(MESSENGER_TRANSPORT_DSN)%'

        routing:
            # Route your messages to the transports
            'App\Lib\MyClass': [amqp, amqp2]
```
Exception:
```
Symfony\Component\DependencyInjection\Exception\ServiceNotFoundException:
Service ".messenger.chain_sender.App\Lib\MyClass" not found: the container inside "Symfony\Component\Messenger\Asynchronous\Routing\SenderLocator" is a smaller service locator that only knows about the "amqp", "amqp2", "messenger.transport.amqp" and "messenger.transport.amqp2" services.

  at vendor/symfony/dependency-injection/ServiceLocator.php:51
  at Symfony\Component\DependencyInjection\ServiceLocator->get('.messenger.chain_sender.App\\Lib\\MyClass')
     (vendor/symfony/messenger/Asynchronous/Routing/SenderLocator.php:38)
  at Symfony\Component\Messenger\Asynchronous\Routing\SenderLocator->getSenderForMessage(object(MyClass))
     (vendor/symfony/messenger/Asynchronous/Middleware/SendMessageMiddleware.php:47)
  at Symfony\Component\Messenger\Asynchronous\Middleware\SendMessageMiddleware->handle(object(Envelope), object(Closure))
     (vendor/symfony/messenger/MessageBus.php:74)
  at Symfony\Component\Messenger\MessageBus->Symfony\Component\Messenger\{closure}(object(Envelope))
     (vendor/symfony/messenger/Middleware/LoggingMiddleware.php:36)
  at Symfony\Component\Messenger\Middleware\LoggingMiddleware->handle(object(MyClass), object(Closure))
     (vendor/symfony/messenger/MessageBus.php:74)
  at Symfony\Component\Messenger\MessageBus->Symfony\Component\Messenger\{closure}(object(MyClass))
     (vendor/symfony/messenger/MessageBus.php:47)
  at Symfony\Component\Messenger\MessageBus->dispatch(object(MyClass))
     (vendor/symfony/messenger/TraceableMessageBus.php:37)
  at Symfony\Component\Messenger\TraceableMessageBus->dispatch(object(MyClass))
     (src/Controller/MyController.php:21)
  at App\Controller\MyController->MyAction(object(TraceableMessageBus))
     (vendor/symfony/http-kernel/HttpKernel.php:149)
  at Symfony\Component\HttpKernel\HttpKernel->handleRaw(object(Request), 1)
     (vendor/symfony/http-kernel/HttpKernel.php:66)
  at Symfony\Component\HttpKernel\HttpKernel->handle(object(Request), 1, true)
     (vendor/symfony/http-kernel/Kernel.php:188)
  at Symfony\Component\HttpKernel\Kernel->handle(object(Request))
     (public/index.php:37)
```

**Comment by sroze** [comment]
Good catch. The following PR should fix it: #28014.
